app/assets/javascripts/components/SlotPicker.es6

```diff
@@ -0,0 +1,119 @@
+      this.selectedSelector = `.${this.selectedClass}`;
+      this.previousMonthSelector = '.js-slot-picker-prev-month';
+      this.nextMonthSelector = '.js-slot-picker-next-month';
+      this.dayButtonSelector = '.js-day-button';
+      this.timesSelector = '.js-times';
+      this.timesHeaderSelector = '.js-slot-picker-header';
+      this.stepSelector = '#telephone_appointment_step';
+      this.monthSelector = '.js-slot-picker-month';
+      this.errorHeadingSelector = '#error-times-heading';
+      this.slotPickerErrorSelector = '#slot-picker-error';
+      this.slotPickerAjaxLoaderSelector = '#slot-picker-ajax-loader';
+
+      this.init();
+    }
+
+    init() {
+      this.revealMonthButtons();
+      this.selectInitialMonth();
+      this.bindEvents();
+    }
+
+    revealMonthButtons() {
+      $(this.previousMonthSelector).removeClass('hidden');
+      $(this.nextMonthSelector).removeClass('hidden');
+    }
+
+    bindEvents() {
+      $(document).on('click', this.previousMonthSelector, this.handlePrevClick.bind(this));
+      $(document).on('click', this.nextMonthSelector, this.handleNextClick.bind(this));
+      $(document).on('click', this.dayButtonSelector, this.handleDayClick.bind(this));
+    }
+
+    handleDayClick(e) {
+      const $selectedButton = $(e.currentTarget),
+        $currentSelectedButton = this.getSelectedButton(),
+        date = $selectedButton.val();
+
+      e.preventDefault();
+
+      $currentSelectedButton.removeClass(this.selectedClass).attr('disabled', false).attr('aria-pressed', false);
+      $selectedButton.addClass(this.selectedClass).attr('disabled', true).attr('aria-pressed', true);
+
+      this.timeout = setTimeout(() => {
+          $(this.timesSelector).html($(this.slotPickerAjaxLoaderSelector).html());
+        }, 200);
+
+      $.post(
+        this.$form.attr('action'),
+        {
+          'telephone_appointment[step]': 2,
+          'telephone_appointment[selected_date]': date,
+          'authenticity_token': this.$form.find('input[name=authenticity_token]').val()
+        }
+      ).done(this.displayTimes.bind(this))
+      .fail(this.displayError.bind(this));
+    }
+
+    displayTimes(data) {
+      clearTimeout(this.timeout);
+      $(this.timesSelector).html(data);
+      $(this.timesHeaderSelector).attr('tabindex', -1).focus().removeAttr('tabindex');
+      $(this.stepSelector).val(2);
+    }
+
+    displayError() {
+      $(this.timesSelector).html($(this.slotPickerErrorSelector).html());
+
+      $(this.errorHeadingSelector).focus().removeAttr('tabindex');
+    }
+
+    selectInitialMonth() {
+      const $selected = this.getSelectedButton(),
+        $month = $selected.parents(this.monthSelector);
+
+      if ($selected.length) {
+        this.selectMonth($month.index());
+      } else {
+        this.selectMonth(0);
+      }
+    }
+
+    selectMonth(index) {
+      this.$months.hide();
+      $(this.$months[index]).show();
```

Comment:
You could save the double lookup here:

`this.$months.hide().eq(index).show()`